test/test_torch.py

```diff
@@ -7661,6 +7661,45 @@ def test_copy_many_to_one(self):
         # storage to a single storage would cause RuntimeError to be thrown
         self.assertRaises(RuntimeError, lambda: torch.zeros(1, 6).expand(5, 6).copy_(torch.zeros(5, 6)))
 
+    def test_copy_float16(self):
+        # Check that fbgemm code no longer reads memory out of bounds, see
+        # copy_impl and fbgemm::Float16ToFloat_ref.
+        # https://github.com/pytorch/pytorch/issues/88543
+
+        # Types to test different code paths in copy_impl.
+        dtypes = (
+            # out_dtype, src_dtype, is_fbgemm
+            (torch.float32, torch.float16, True),  # fbgemm
```

Comment:
Note: `is_fbgemm/ok_fbgemm` here mean "can be handled by fbgemm" rather than "will be". Some of the `float16` cases will be handled by TI if the `fbgemm` `if` in `copy_impl` fails to match. I've checked which ones are handled by adding a print on the C++ side. There's no way to assert it on the Python side, so this is just for information purposes.